src/components/baseline/components/ImageList.js

```diff
@@ -0,0 +1,151 @@
+import React, { Component } from "react";
+import classNames from "classnames";
+import { ImageViewer } from "./ImageViewer";
+import { Icon } from "components/baseline";
+import {
+  SortableContainer,
+  SortableElement,
+  arrayMove
+} from "react-sortable-hoc";
+
+export const ImageListItem = SortableElement(({
+  content,
+  index,
+  handleClick
+}) => {
+  const src = !content.url ? content.preview : content.url;
+
+  return (
+    <div className="ImageListItem" key={index}>
+      <div className="ImageListItem-card">
+        <div className="image">
+          <img src={src} alt={content.name} />
+          <button
+            className="Image-popoutButton viewTarget"
+            onClick={handleClick}
+          >
+            <Icon name="popout" size={20} />
+          </button>
+        </div>
+        <textarea
+          value={content.caption}
+          placeholder="Enter description..."
+          onChange={({ target }) =>
+            console.log("TODO handle caption save", target.value)}
+        />
+      </div>
+    </div>
+  );
+});
+
+const ImageGrid = SortableContainer(({ items, className, handleClick }) => {
+  const listItems = items.map((item, index) => (
+    <ImageListItem
+      key={`imageListItem${index}`}
+      content={item}
+      index={index}
+      handleClick={() => handleClick(index)}
+    />
+  ));
+  return <div className={className}>{listItems}</div>;
+});
+
+class ImageList extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      selection: null,
+      showViewer: false,
+      items: props.content
+    };
+  }
+
+  componentWillReceiveProps({ content }) {
+    this.setState({ items: content });
+  }
+
+  handleClick = selection => this.setState({ selection, showViewer: true });
+
+  onSortEnd = ({ oldIndex, newIndex }) =>
+    this.props.onReorder(arrayMove(this.state.items, oldIndex, newIndex));
+
+  shouldCancelStart = e => {
+    const disabledElements = [
```

Comment:
need to move this into a utility to use with `react-sortable-hoc` this is a more robust solution. Or should submit a PR to `react-sortable-hoc` to take a list of class names to disable dnd behavior as needed.